backend/packages/Upgrade/src/api/services/SegmentService.ts

```diff
@@ -149,7 +164,134 @@ export class SegmentService {
+        if (isSegmentJSONValid) {
+          parsedJsonData.push(segmentTemp);
+        } else {
+          importFileErrors.push({
+            fileName: fileName,
+            error: 'Invalid Segment JSON data ' + this.missingAllProperties,
+          });
+          return;
+        }
+      });
+      const [importedSegments] = await Promise.all([this.importJsonSegments(parsedJsonData, logger)]);
+      return { segments: importedSegments, importErrors: importFileErrors };
+    } else {
+      const [importedSegments] = await Promise.all([this.importCsvSegments(segments, logger)]);
+      return { segments: importedSegments, importErrors: importFileErrors };
+    }
+  }
+
+  private isJSONString(data: any): boolean {
+    try {
+      JSON.parse(data);
+      return true;
+    } catch (error) {
+      return false;
+    }
+  }
+
+  validateSegmentJSON(segment: SegmentInputValidator): boolean {
+    const segmentSchema: Record<keyof any, string> = {
+      id: 'string',
+      name: 'string',
+      context: 'string',
+      description: 'string',
+      userIds: 'array',
+      groups: 'interface',
+      subSegmentIds: 'array',
+      type: 'enum',
+    };
+
+    this.missingAllProperties = this.checkForMissingProperties({ schema: segmentSchema, data: segment });
+
+    if (this.missingAllProperties.length > 0) {
+      return false;
+    } else {
+      return this.missingAllProperties.length === 0;
+    }
+  }
+
+  private checkForMissingProperties(segmentJson: ImportSegmentJSON) {
+    const { schema, data } = segmentJson;
+    const missingProperties = Object.keys(schema)
+      .filter((key) => data[key] === undefined)
+      .map((key) => key as keyof SegmentInputValidator)
+      .map((key) => `${key}`);
+    return missingProperties.join(', ');
+  }
+
+  public async importCsvSegments(segments: SegmentFile[], logger: UpgradeLogger): Promise<Segment[]> {
+    const csvData: string[] = [];
+    const fileNames: string[] = [];
+
+    segments.forEach((segment) => {
+      csvData.push(segment.fileContent);
+      const filename = segment.fileName.split('.');
+      fileNames.push(filename[0]);
+    });
+
+    const segmentData = csvData.map((data, index) => {
+      const rows = data.replace(/"/g, '').split('\n');
+
+      let segmentContext: string;
+      const segmentUserIds: string[] = [];
+      const subSegmentIds: string[] = [];
+      const segmentGroups: Group[] = [];
+
+      for (let i = 1; i < rows.length; i++) {
+        const row = rows[i];
+        const rowValues = row.split(',');
+
+        // Extract the ID
+        const id = rowValues[0];
+        if (!id) {
+          continue;
+        }
+
+        const memberType = rowValues[1].trim().toLowerCase();
+        if (memberType === 'individual') {
+          segmentUserIds.push(id);
+        } else if (memberType === 'segment') {
+          subSegmentIds.push(id);
+        } else if (memberType === 'context') {
+          segmentContext = id;
+        } else {
```

Comment:
You should check the `memberType` of `group` before pushing it into the `segmentGroups`

Reply:
Don't we need any validation for the data coming from CSV? Like context should exist and other?

Reply:
these validations are added in importJsonSegment as it didn't had these so wrong context or group name segments were getting created.